lib/android/fido/state.dart

```diff
@@ -165,26 +166,152 @@ final androidFingerprintProvider = AsyncNotifierProvider.autoDispose
         _FidoFingerprintsNotifier.new);
 
 class _FidoFingerprintsNotifier extends FidoFingerprintsNotifier {
+  final _events = const EventChannel('android.fido.fingerprints');
+  late StreamSubscription _sub;
+
   @override
   FutureOr<List<Fingerprint>> build(DevicePath devicePath) async {
-    return [];
+    _sub = _events.receiveBroadcastStream().listen((event) {
+      final json = jsonDecode(event);
+      if (json == null) {
+        state = const AsyncValue.loading();
+      } else {
+        List<Fingerprint> newState = List.from(
+            (json as List).map((e) => Fingerprint.fromJson(e)).toList());
+        state = AsyncValue.data(newState);
+      }
+    }, onError: (err, stackTrace) {
+      state = AsyncValue.error(err, stackTrace);
+    });
+
+    ref.onDispose(_sub.cancel);
+    return Completer<List<Fingerprint>>().future;
   }
 
   @override
   Stream<FingerprintEvent> registerFingerprint({String? name}) {
     final controller = StreamController<FingerprintEvent>();
+    const registerEvents = EventChannel('android.fido.registerFp');
+
+    final registerFpSub =
+        registerEvents.receiveBroadcastStream().skip(1).listen((event) {
```

Comment:
What is the event we're skipping here?

Reply:
We decided to refactor this later, the backing live data is not the best way to solve the fingerprint registration, but using skip(1) works.